### PR TITLE
Updated Selector to be a user friendly ObjectSelector

### PR DESCRIPTION
**Pull request by jlstevens**
This is nice and convenient in most cases with a shorter name. Addresses the second item listed in https://github.com/pyviz/datashader/issues/698.

### CODE REVIEW

**Comment by jlstevens** [comment]
Note: The tests are currently failing due to issues submitting coverage reports to coveralls which is unrelated to this PR.

File: param/__init__.py [hunk 1]
```diff
@@ -1114,7 +1114,40 @@ def get_range(self):
...
+
+
+        py3_ordered_dicts = (sys.version_info.major == 3) and (sys.version_info.minor >= 6)
+        vanilla_odicts = (sys.version_info.major > 3) or py3_ordered_dicts
+
+        is_odict = isinstance(objects, (OrderedDict)) or (vanilla_odicts and isinstance(objects, dict))
+        if is_odict:
+            autodefault = list(objects.values())[0]
+        elif isinstance(objects, dict):
+            main.param.warning("Dictionaries prior to Python 3.6 are not ordered. Using random default.")
```
[the remark this answers is not in the document]
Comment (jlstevens) [reply param/__init__.py]: Addressed in 15d4846

File: param/__init__.py [hunk 2]
```diff
@@ -1114,7 +1114,40 @@ def get_range(self):
...
+    argument which sufficient in many common use cases.
+    """
+    def __init__(self,objects=None, default=None, instantiate=False,
+                 compute_default_fn=None,check_on_set=None,allow_None=None,**params):
+
+
+        py3_ordered_dicts = (sys.version_info.major == 3) and (sys.version_info.minor >= 6)
+        vanilla_odicts = (sys.version_info.major > 3) or py3_ordered_dicts
+
+        is_odict = isinstance(objects, (OrderedDict)) or (vanilla_odicts and isinstance(objects, dict))
```
[the remark this answers is not in the document]
Comment (jlstevens) [reply param/__init__.py]: Addressed in df609bd

File: param/__init__.py [hunk 3]
```diff
@@ -87,6 +87,16 @@ def as_unicode(obj):
     return unicode(obj)
 
 
+def is_ordered_dict(d):
+    """
+    Predicate checking for ordered dictionaries. OrderedDict is always
+    ordered, an vanilla Python dictionaries are ordered for Python 3.6+
```
Comment (jbednar) [inline param/__init__.py]:
```suggestion
    ordered, and vanilla Python dictionaries are ordered for Python 3.6+
```

**Pull request merged**